src/hotspot/share/jfr/periodic/sampling/jfrThreadSampler.cpp

```diff
@@ -386,6 +387,11 @@ bool JfrThreadSampleClosure::do_sample_thread(JavaThread* thread, JfrStackFrame*
     return false;
   }
 
+  // Process the oops in the thread head before calling into code that wants to
+  // stack walk over Loom continuations. The stack walking code will otherwise
+  // skip frames in stack chunks on the Java heap.
+  StackWatermarkSet::start_processing(thread, StackWatermarkKind::gc);
```

Comment:
Maybe this should be moved to under each thread_state_in_java() and thread_state_in_native() preliminary check, see below. That way we only call on threads that we will eventually suspend.?

Reply:
Just put it in each of JfrThreadSampler::sample_thread_in_java() and JfrThreadSampler::sample_thread_in_native() member functions,

Reply:
Pushed a commit. Could you check it?